### PR TITLE
Video capture module

### PR DESCRIPTION
**Pull request by pawtom**
Extend the Squape package with a video module. The main feature is allowing capturing videos using a context manager with the option to automatically delete a video when execution was successful.  This is a space saver if we are only interested in videos to analyze failure in the tests. 

For discussion: Should the video file be deleted or kept as an empty file?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ pip install squape
 The package consists of several modules:
 - report - adds features to enhance the reporting and verification capabilities of Squish.
 - settings - allows using various test settings as context managers
+- video - allows capturing videos using a context manager with option to automatically delete a video when execution was successful
```
Comment (Adakar) [inline README.md:21]: The description should be more generic as more functionalities may come later. It should provide information about the idea not the implementation

File: squape/video.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, Cyber Alpaca
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import os
+from contextlib import contextmanager
+
+import squishinfo
```
Comment (Adakar) [inline squape/video.py:10]: squishinfo was not used before, thus it's not mentioned in the pre-commit configuration yet.
Please add it into pre-commit-config.yaml file as:
`--unclassifiable-application-module, squishinfo,`

File: squape/video.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        set: The set of filenames with mp4 extension
+    """
+    video_dir = os.path.join(
+        squishinfo.resultDir,
+        squishinfo.testSuiteName,
+        squishinfo.testCaseName,
+        "attachments",
+    )
+
+    videos_list = set()
```
Comment (Adakar) [inline squape/video.py:46]: It is called list but in fact it is a set.
```suggestion
    videos = set()
```

File: squape/video.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    video_dir = os.path.join(
+        squishinfo.resultDir,
+        squishinfo.testSuiteName,
+        squishinfo.testCaseName,
+        "attachments",
+    )
+    for video_name in videos:
+        test.log(f"Remove video: {video_name}")
+        file_name = os.path.join(video_dir, video_name)
+        os.remove(file_name)
```
Comment (Adakar) [inline squape/video.py:73]: what if the file doesn't exist?

File: squape/video.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+def video_capture(message: str, delete_on_success: bool = False) -> None:
+    """Allows using Squish's video capture as context managers.
+    https://doc.qt.io/squish/squish-api.html#test-startvideocapture-message
+    Optionally delete a captured video when the execution was successful (no failures)
+
+    Args:
+        message (str): log a video n the test report using the specified message
+        delete_on_success (bool): Whether to delete captured video
+        when there were no failures. Defaulting to False.
+
```
Comment (Adakar) [inline squape/video.py:86]: Missing examples

File: squape/video.py (hunk 2 above)
Comment (pawtom) [reply squape/video.py:10]: Added with [ad1cf3d](https://github.com/CyberAlpaca/squish-api-python-extension/pull/16/commits/ad1cf3de32521e92df6f1ff55b460d199652dfdb)

File: squape/video.py (hunk 3 above)
Comment (pawtom) [reply squape/video.py]: Fixed with [8bbe6a6](https://github.com/CyberAlpaca/squish-api-python-extension/pull/16/commits/8bbe6a66fbea305d581cca86137158c0f0df0f9e)

File: squape/video.py (hunk 5 above)
Comment (pawtom) [reply squape/video.py:86]: Fixed with [fc35fe1](https://github.com/CyberAlpaca/squish-api-python-extension/pull/16/commits/fc35fe1c84931a49c280f2d718d19537f5129ac2)

File: squape/video.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        test.log(f"Remove video: {video_name}")
+        orig_video = Path(video_dir, video_name)
+        shutil.copy(empty_video, orig_video)
+
+
+@contextmanager
+def video_capture(message: str = None, remove_on_success: bool = False) -> None:
+    """Allows using Squish's video capture as context managers.
+    https://doc.qt.io/squish/squish-api.html#test-startvideocapture-message
+    Optionally remove a captured video when the execution was successful (no failures)
```
Comment (Adakar) [inline squape/video.py:92]: Missing information about video that replaces the original ones

File: squape/video.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+    empty_video = Path(res_container, "empty_video_with_message.mp4")
+
+    for video_name in videos:
+        test.log(f"Remove video: {video_name}")
+        orig_video = Path(video_dir, video_name)
+        shutil.copy(empty_video, orig_video)
+
+
+@contextmanager
+def video_capture(message: str = None, remove_on_success: bool = False) -> None:
```
Comment (Adakar) [inline squape/video.py:89]: Emptry string instead of None as the default for the message would simplify the function

File: squape/video.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+import shutil
+from contextlib import contextmanager
+from importlib import resources
+from pathlib import Path
+
+import squishinfo
+import test
+
+
+def _failures_results_count() -> int:
```
Comment (Adakar) [inline squape/video.py:17]:
```suggestion
def _failure_results_count() -> int:
```

**Review by Adakar — CHANGES_REQUESTED** [review]

File: squape/video.py (hunk 4 above)
Comment (pawtom) [reply squape/video.py]: The comment is outdated as after the latest changes videos are not removed.

File: squape/video.py [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+    if not os.path.exists(video_dir):
+        return videos_set
+
+    for file in os.listdir(video_dir):
+        if file.endswith(".mp4"):
+            videos_set.add(file)
+    return videos_set
+
+
+def _remove_videos(videos: set) -> None:
```
[the remark this answers is not in the document]
Comment (pawtom) [reply squape/video.py]: Fixed with [ddb4fd1](https://github.com/CyberAlpaca/squish-api-python-extension/pull/16/commits/ddb4fd1375ad5720b5e5667caf352c76ede1fff1)

File: squape/video.py (hunk 7 above)
Comment (pawtom) [reply squape/video.py]: Fixed with [95f6c0a](https://github.com/CyberAlpaca/squish-api-python-extension/pull/16/commits/95f6c0a0268b5100a8783db87d1f8d20669c3a76)

File: squape/video.py (hunk 8 above)
Comment (pawtom) [reply squape/video.py]: Fixed [7eb6510](https://github.com/CyberAlpaca/squish-api-python-extension/pull/16/commits/7eb6510f99dddb29996262c402dc84a7f7adbbe2)

**Review by Adakar — APPROVED** [review]
Thanks!

**Pull request merged**